src/executor/query/IndexScanExecutor.cpp

```diff
@@ -28,8 +30,16 @@ folly::Future<Status> IndexScanExecutor::indexScan() {
         DataSet dataSet({"dummy"});
         return finish(ResultBuilder().value(Value(std::move(dataSet))).finish());
     }
+
+    const auto &ictxs = lookup->queryContext();
+    auto iter = std::find_if(
+        ictxs.begin(), ictxs.end(), [](auto &ictx) { return !ictx.index_id_ref().is_set(); });
+    if (ictxs.empty() || iter != ictxs.end()) {
+        return Status::Error("There is no index to use at runtime");
+    }
+
```

Comment:
Why is the index ID not set?

Reply:
Now index is selected in multiple opt rules. when all rules could not select the index, IndexFullScanNode without index is still in the execution plan. so here needs to check the real index at runtime.